### PR TITLE
Feat: add grpc interceptors for event service

### PR DESCRIPTION
**Pull request by Nhsdkk**
1) Add custom interceptor
2) Add database connection termination on server stop
3) Fix error when auth returned user when he already logged off
4) Add permission denied error

### CODE REVIEW

File: identity_service/main.py [hunk 1]
```diff
@@ -1,3 +1,4 @@
+from asyncio.exceptions import CancelledError
```
Comment (MaxPlays35) [inline identity_service/main.py]: Extra import

File: identity_service/main.py [hunk 2]
```diff
@@ -33,8 +35,18 @@ async def serve() -> None:
     await server.wait_for_termination()
 
 
+async def handle_serve_error() -> None:
+    """Handle server stop"""
+    try:
+        await serve()
+    except CancelledError:
```
Comment (MaxPlays35) [inline identity_service/main.py]: use `asyncio.CancelledError`

**Review by MaxPlays35 — CHANGES_REQUESTED** [review]

File: identity_service/main.py (hunk 1 above)
Comment (Nhsdkk) [reply identity_service/main.py]: fixed

File: identity_service/main.py (hunk 2 above)
Comment (Nhsdkk) [reply identity_service/main.py]: fixed

**Review by MaxPlays35 — APPROVED** [review]

**Review by ZixMai — APPROVED** [review]

**Pull request merged**